### PR TITLE
[4.0] WebAssetManager: Speed up enable/disable asset

### PR DESCRIPTION
**Pull request by Fedik**
### Summary of Changes

WebAssetManager calculate dependencies on "request" (when it really need) instead of each call of `enableAsset`/`disableAsset`.
This will speed up operations like:
```php
$wa->enableAsset('foo')
  ->enableAsset('bar')
  ->disableAsset('bear')
  ->enableAsset('beer');
```

Additionally I have updated a comment for a Manager interface, added missed throws tags.

### Testing Instructions
Apply the patch and make sure everything still work :wink: 

ping @wilsonge 

for reference #22435

### CODE REVIEW

**Comment by Fedik** [comment]
I think I am banned to see ci.joomla.org, so cannot help with Drone, sorry :smile:

**Comment by dgrammatiko** [comment]
@Fedik here is the failure:
```
FILE: ...github.com/joomla/joomla-cms/libraries/src/WebAsset/WebAssetManager.php
FOUND 1 ERROR(S) AFFECTING 1 LINE(S)
 91 | ERROR | Expected 1 blank line before member var; 2 found
```

**Comment by Fedik** [comment]
@dgrammatiko thanks!

File: libraries/src/WebAsset/WebAssetManager.php [hunk 1]
```diff
@@ -80,6 +80,16 @@ class WebAssetManager implements WebAssetManagerInterface, DispatcherAwareInterf
 	 */
 	protected $useVersioning = true;
 
+
```
Comment (dgrammatiko) [inline libraries/src/WebAsset/WebAssetManager.php:93]: remove empty line

**Review by dgrammatiko — CHANGES_REQUESTED** [review]

File: libraries/src/WebAsset/WebAssetManager.php (hunk 1 above)
Comment (Fedik) [reply libraries/src/WebAsset/WebAssetManager.php:93]: ok now

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**